manager/integration/tests/test_settings.py

```diff
@@ -896,25 +896,28 @@ def test_setting_concurrent_rebuild_limit(client, core_api, volume_name):  # NOQ
         volume1 = client.by_id_volume(volume1_name)
         volume2 = client.by_id_volume(volume2_name)
 
-        if volume1.rebuildStatus == [] and \
-                volume2.rebuildStatus == [] and \
-                rebuild_started is False:
-            continue
-        elif volume1.rebuildStatus == [] and \
-                volume2.rebuildStatus == [] and \
-                rebuild_started is True:
+        if (volume1.rebuildStatus == [] or volume2.rebuildStatus == []) \
+                and rebuild_started:
             break
-        elif volume2.rebuildStatus == []:
-            assert volume1.rebuildStatus[0].state == "in_progress"
-            rebuild_started = True
-        elif volume1.rebuildStatus == []:
-            assert volume2.rebuildStatus[0].state == "in_progress"
-            rebuild_started = True
+
+        # Check volume1 if volume2 is not rebuilding
+        if not volume2.rebuildStatus:
+            if volume1.rebuildStatus:
+                assert volume1.rebuildStatus[0].state == "in_progress"
+                rebuild_started = volume1.rebuildStatus[0].progress >= 50
+
+        # Check volume2 if volume1 is not rebuilding
+        if not volume1.rebuildStatus:
```

Comment:
Will a volume doesn't have `rebuildStatus`?

Reply:
Yes, according to [the distribution scenario mentioned](https://github.com/longhorn/longhorn-tests/blob/v1.3.x/manager/integration/tests/test_settings.py#L783-L786), one of the volumes does not have a `rebuildStatus`.